chain/ethereum.ts

```diff
@@ -103,11 +103,12 @@ export namespace ethereum {
     }
 
     toMatrix(): Array<Array<Value>> {
-      assert(
-        this.kind == ValueKind.ARRAY || this.kind == ValueKind.FIXED_ARRAY,
-        'Ethereum value is not an array (matrix conversion).',
-      )
-      return changetype<Array<Array<Value>>>(this.data as u32)
+      let valueArray = this.toArray()
```

Comment:
Just to note, this conversion is needed to first convert what's a `Value { kind: ARRAY, data: i32 }` to an `Array<Value>` so we can later convert each `Value` into an `Array<Value>` too 🙂 

It isn't possible to just do a `changetype` with a recursive structure like this.